src/importer/sources/desktop_source.py

```diff
@@ -0,0 +1,215 @@
+# desktop_source.py
+#
+# Copyright 2023 kramo
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import shlex
+from hashlib import sha3_256
```

Comment:
Not because of security, but hash collisions and better forwards compatibility. Using a legacy algorithm for a ~20% speed increase seems silly. I also specify the `usedforsecurity=False` arg.